File: backend/api/poster/entity/poster.go
package entity

import crudModel "gitlab.universedigital.my.id/library/golang/crud/model"

type PosterLive struct {
	crudModel.AuditTrail
	DeceasedName      string `gorm:"column:deceased_name" json:"deceasedName"`
	Title             string `gorm:"column:title" json:"title"`
	Keterangan        string `gorm:"column:keterangan" json:"keterangan"`
	Age               string `gorm:"column:age" json:"age"`
	DateOfPassing     string `gorm:"column:date_of_passing" json:"dateOfPassing"`
	TimeOfPassing     string `gorm:"column:time_of_passing" json:"timeOfPassing"`
	TimeOfPassingZone string `gorm:"column:time_of_passing_zone" json:"timeOfPassingZone"`
	ImageURL          string `gorm:"column:image_url" json:"imageUrl"`
	PlaceOfPassing    string `gorm:"column:place_of_passing" json:"placeOfPassing"`
	PlaceLaidOut      string `gorm:"column:place_laid_out" json:"placeLaidOut"`
	ProcessionType    string `gorm:"column:procession_type" json:"processionType"`
	ProcessionDate    string `gorm:"column:procession_date" json:"processionDate"`
	ProcessionTime    string `gorm:"column:procession_time" json:"processionTime"`
	ProcessionTZ      string `gorm:"column:procession_time_zone" json:"processionTimeZone"`
	ProcessionPlace   string `gorm:"column:procession_place" json:"processionPlace"`
	MessageFrom       string `gorm:"column:message_from" json:"messageFrom"`
	Theme             string `gorm:"column:theme" json:"theme"`
	FamilyData        string `gorm:"column:family_data" json:"familyData"`
	CondolenceMessage string `gorm:"column:condolence_message" json:"condolenceMessage"`
	RelationSummary   string `gorm:"column:relation_summary" json:"relationSummary"`
	DeathStatement    string `gorm:"column:death_statement" json:"deathStatement"`
}

func (PosterLive) TableName() string {
	return "poster"
}

type PosterHis struct {
	PosterLive
}

func (PosterHis) TableName() string {
	return "poster_his"
}

File: frontend/src/types/poster.ts
export interface PosterFormData {
  deceasedName: string
  title: string
  keterangan: string
  age: string
  dateOfPassing: string
  /** HH:mm (24h), opsional */
  timeOfPassing: string
  /** WIB | WITA | WIT */
  timeOfPassingZone: string
  imageUrl: string
  placeOfPassing: string
  placeLaidOut: string
  processionType: string
  processionDate: string
  processionTime: string
  /** WIB | WITA | WIT */
  processionTimeZone: string
  processionPlace: string
  messageFrom: string
  theme: string
  familyData: string
  condolenceMessage: string
  relationSummary: string
  deathStatement: string
}

export const DEFAULT_CONDOLENCE_MESSAGE =
  'Kami turut berduka cita atas kepergianny. Semoga almarhum/almarhumah diterima di sisi-Nya dan keluarga yang ditinggalkan diberikan ketabahan.'

export type PosterTextField =
  | 'deceasedName'
  | 'title'
  | 'age'
  | 'dateOfPassing'
  | 'placeOfPassing'
  | 'messageFrom'
  | 'condolenceMessage'

export type PosterFontFamily = 'serif' | 'sans-serif' | 'monospace'
export type PosterFontStyle = 'normal' | 'italic'

export interface PosterTextStyle {
  fontSize: number
  fontFamily: PosterFontFamily
  fontStyle: PosterFontStyle
  color: string
  fontWeight: 400 | 500 | 600 | 700 | 800
  textShadow: boolean
}

export type PosterTypographySettings = Record<PosterTextField, PosterTextStyle>
export type PosterTheme = 'elegant-night' | 'warm-classic' | 'soft-gray'
export type EditorView = 'list' | 'create' | 'template'

export interface PosterTemplatePosition {
  x: number
  y: number
}

export type PosterTemplateLayout = Record<PosterTextField, PosterTemplatePosition>

export interface CustomHeadline {
  id: string
  text: string
  x: number
  y: number
  fontSize: number
  color: string
  fontWeight: 400 | 500 | 600 | 700 | 800
  locked: boolean
}

export type TemplateBackgroundType = 'theme' | 'image'

export interface TemplateDesign {
  id: string
  name: string
  headerMode: 'recid' | 'headline'
  headerText: string
  theme: PosterTheme
  backgroundType: TemplateBackgroundType
  backgroundValue: string
  layout: PosterTemplateLayout
  headlines: CustomHeadline[]
  createdDateTime?: string
  updateDateTime?: string
}

export interface PosterRecord extends PosterFormData {
  id: string
  createdAt: string
  createdDateTime?: string
  updateDateTime?: string
}

export interface APIResponse<TData> {
  message: string
  data: TData
}
